### PR TITLE
Ncnews 7

### PR DESCRIPTION
**Pull request by rangararazin**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -155,3 +155,69 @@ describe("GET: /api/articles/:article_id/comments", () => {
       });
   });
 });
+
+describe("POST : /api/articles/:article_id/comments", () => {
```
Comment (JVal-NC) [inline __tests__/app.test.js]: Good tests but would benefit from the following additional tests:
test for endpoint ignoring unnecessary body object properties,
test for non existent ID, e.g. 0 or 9999,

File: __tests__/app.test.js [hunk 2]
```diff
@@ -155,3 +155,69 @@ describe("GET: /api/articles/:article_id/comments", () => {
...
+          article_id: 1,
+          votes: 0,
+          created_at: expect.any(String),
+          author: "rogersop",
+          body: "lorem ipsum roger",
+        });
+      });
+  });
+
+  test("400:if user doesn't exist", () => {
```
Comment (JVal-NC) [inline __tests__/app.test.js]: This would be a status 404

File: utils/db.js [hunk 3]
```diff
@@ -14,3 +15,17 @@ WHERE article_Id =$1`,
       }
     });
 };
+
+exports.checkUserExist = (table, column, value) => {
```
Comment (JVal-NC) [inline utils/db.js]: I like this but seems to be kept generic but has name and promise reject specific to users, can this either be generalised or be amended so it is specific to users?

**Review by JVal-NC — COMMENTED** [review]
See review comments for required changes

File: utils/db.js (hunk 3 above)
Comment (rangararazin) [reply utils/db.js]: Yes made it specifically for user

File: __tests__/app.test.js [hunk 4]
```diff
@@ -155,3 +155,85 @@ describe("GET: /api/articles/:article_id/comments", () => {
...
+    return request(app)
+      .post("/api/articles/45454")
+      .send({
+        username: "rogersop",
+        body: "lorem ipsum roger",
+      })
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Article not found");
+      });
```
Comment (sirKevlar) [inline __tests__/app.test.js:237]: Nearly there, but missing tests: It's possible for username and body (the keys) to be misspelled

**Comment by Liam310** [comment]
Looks like you're good to go now Razin! 👍

**Pull request merged**